Issue # #524 - Combine locations.json and timeseries json

firebuggirl opened the pull request:
Hi all,

I saw this issue posted earlier today and thought that I'd take a crack at it. I re-assigned the destructured values `locations, timeseriesByDate` in `src/shared/timeseries.js` and re-assigned them to a single output variable so as to output one JSON file `timeseries-locations.json`. I'm not sure if the data is being output in the order that is needed, but all key/values have been output to a single file.

Hope this helps!

Cheers:-)

firebuggirl committed:
docker commands

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl committed:
Merge github.com:lazd/coronadatascraper

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl committed:
delete Docker commands from master branch

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl committed:
Issue #524 Combine locations.json and timeseries.json into a single file

lazd commented:
Ok awesome, I say we keep `locations.json` still, and we just call the output file `timeseries.json` (same filename as before, but now with different fields). 

Think that'll upset the masses?

lazd commented on src/shared/timeseries.js:
```diff
@@ -226,9 +226,11 @@ async function generateTimeseries(options = {}) {
-  await fs.writeFile(path.join('dist', `timeseries.json`), JSON.stringify(timeseriesByDate, null, 2));
-  await fs.writeFile(path.join('dist', `locations.json`), JSON.stringify(locations, null, 2));
-
+  const combineLocationsTimeseriesByDate = { locations, timeseriesByDate };
```
Maybe the prop should just be `timeseries` inside of here? Or maybe `dates`?

Thinking about how I'd iterate over it:

```js
const timeseries = getJSON('timeseries.json);
for (let date of timeseries.dates) {
  for (let locationId of date) {
    let location = timeseries.locations[locationId]
    // ... do stuff
  }
}
```

That doesn't feel great either. Thoughts?

firebuggirl replied:
"Think that'll upset the masses?"

Yikes... I can't say for sure:-)

firebuggirl replied:
> Maybe the prop should just be `timeseries` inside of here? Or maybe `dates`?
> 
> Thinking about how I'd iterate over it:
> 
> ```js
> const timeseries = getJSON('timeseries.json');
> for (let date of timeseries.dates) {
>   for (let locationId of date) {
>     let location = timeseries.locations[locationId]
>     // ... do stuff
>   }
> }
> ```
> 
> That doesn't feel great either. Thoughts?

..thinking....it seems like it would/could be a cleaner approach:-)

lazd replied:
I wonder if others have ideas for the names of the props? @chunder @qgolsteyn @camjc @ryanblock @NateBaldwinDesign @JacobDFrank?

camjc replied:
The above looks pretty good to me.
Would agree that the original name `combineLocationsTimeseriesByDate` isn't optimal anyhow; mainly because it feels like a function thanks to its first word being a verb.

firebuggirl replied:
I see your point @camjc :-)

hyperknot commented:
I'm not sure I understand the original problem here. Why isn't timeseries-byLocation.json good as it is? I agree we should remove timeseries.json in current form and replace it with timeseries-byLocation.json.

To do this, we'd need to:
- use ids everywhere. So `New South Wales, AUS` becomes `iso2:AU-NSW`.
- always use JSON objects, not lists
- no almost-random integer numbers

lazd commented:
`timeseries-byLocation.json` is totally hard to work with for my use case, i like `timeseries.json` + `locations.json`. This PR combines them so it's more obvious how they're used together.

lazd commented:
If we index `locations.json` by the `iso2:AU-NSW` instead of `234` or whatever it is, that's fine, but these files would still need to be combined to be easier to understand.

camjc commented on src/shared/timeseries.js:
```diff
@@ -226,9 +226,11 @@ async function generateTimeseries(options = {}) {
+  const combineLocationsTimeseriesByDate = { locations, timeseriesByDate };
+  await fs.writeFile(
+    path.join('dist', `timeseries-locations.json`),
+    JSON.stringify(combineLocationsTimeseriesByDate, null, 2));
```
You'll have the happiest time with prettier if you set up your IDE to auto-fix eslint on-save (obviously up to you :) )

hyperknot commented:
> timeseries-byLocation.json is totally hard to work with for my use case, i like timeseries.json + locations.json.

Can you explain why? I find timeseries-byLocation.json by far the best to work with.

lazd commented:
@hyperknot it's a pain in the ass for what I did at http://blog.lazd.net/coronaglobe -- I want to iterate over dates only and rip out locations from there.

Regardless, that's beside the point. `timeseries.json` could be made more consumable, and this PR does that.

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

jzohrab commented:
Hi @firebuggirl , thanks very much for the PR!  Things in the code have changed since you opened this.  If you are still available, can you fix the conflicts, and make sure that lint and the tests pass? (`yarn lint` and `yarn test`)  As @lazd mentioned, this change makes the data more consumable, which is great.  Cheers! jz

lazd commented:
Yeah I'm keen to merge this one. I think the file should be:

```
{
  locations: {},
  timeseries: {}
}
```

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl commented:
Hi @jzohrab & @lazd ,

I am just seeing this now. I have just pulled from the upstream. Let me get acquainted with what's changed and see what I can do. I'll get back soon.
 
Cheers:-)

firebuggirl commented:
OK..

This is what I can say is happening so far (after having pulled from the upstream on the master branch + switching to this branch and running `git pull origin master` + running `yarn timeseries` again..):

I am getting no errors when running `yarn test` + `yarn lint` (actually, there were a couple of issues with prettier not liking my syntax/layout..fixed that..).  

I did change the constant variable name fro `combineLocationsTimeseriesByDate` to `LocationsAndDates` (let me know if that works...).

As for any further iterations, I'm not quite sure what to do at this point. I am willing to study the code more if need be (& I'm open to directions & instructions if need be). This is a big project with lots of moving parts and I don't want to make any silly mistakes:-)

Let me know if I should re-push the existing code &/or add something new.

Cheers!

firebuggirl commented:
I'll check back in the morning..(California time). Be well everyone:-)

firebuggirl commented:
Hi all,

...just checking in... I have just run into some issues with testing...strange because I did not encounter any errors last night after running `yarn lint` and `yarn test`.  I did run `yarn timeseries` & `yarn start` again and as a result there are tests failing for some of the scrapers.. I'm not sure what's going on with that...be back in a bit:-)

lazd commented:
@jzohrab can you help @firebuggirl with tests? I think it might be false positive.

jzohrab commented:
Hi @firebuggirl - thanks for keep us updated!  Are you on our Slack?  It might be easier to chat on there.  My Slack name is the same as my github.  "Tests failing for some scrapers" could be false negatives -- i.e., nothing to do with your changes -- we're fixing tests and methods.

firebuggirl commented:
Hi @lazd, 

I just realized that I had the wrong version of Node running after restarting my computer this morning. All is good now. All tests are passing. I should just set my default version to 13:-)

firebuggirl committed:
yarn lint...fix errors

firebuggirl committed:
yarn build

firebuggirl committed:
Merge branch 'master' of github.com:lazd/coronadatascraper

firebuggirl committed:
update from upstream

firebuggirl committed:
complete merge of site/index.html & src/shared/timeseries/index.js

firebuggirl committed:
get latest upstream 4/8/2020 => add LocationsAndDates to src/shared/timeseries/index.js

firebuggirl commented:
Hi again all, 

I deleted a couple of my previous comments that are no longer relevant so as not to clutter this space up too much. I noticed that there have been some changes since last night, so I have updated everything (today => 3/8/20) with the current master branch (along with the changes that were proposed earlier).

Hope this helps:-)

jzohrab commented:
@lazd , I was chatting with @firebuggirl today on Slack and we talked about testing and stability.  If this works, we can add it, and then backlog adding some testing for it (e.g., just a smoke test for this, to ensure it actually runs at all points in the future).  Thoughts?  What else might be needed for this PR to be complete?

Thanks again @firebuggirl !  z

jzohrab commented:
Per discussion today, @lazd will rebase this against master and will merge it shortly.

firebuggirl commented:
Thanks for the update @jzohrab !

jzohrab commented:
Updating @firebuggirl - we're dealing with some architecture redesign at the moment, so I'm not merging this for now.  Will keep it open so we can merge later.  Don't worry about fixing conflicts etc.  Cheers! z

firebuggirl commented:
Hi @jzohrab ,

Sounds good. Thanks for letting me know! I was wondering...Cheers:-)

lazd committed:
Merge branch 'master' into combine-locations-timeseries-json

lazd committed:
fix: rename output

jzohrab commented:
Hi @firebuggirl -- after a _long_ time sitting on this branch I'm going to close this PR.  Apologies for the time sink, this isn't ideal, but that's how things worked out.  When you submitted this, we were just starting moving to a new project (Li, https://github.com/covidatlas/li), and I didn't want to merge this b/c we were focused on that.  Now we're getting close to the end of the migration, and this idea has, for the moment, become obsolete.  We're generating reports in Li (https://github.com/covidatlas/li/blob/master/docs/reports.md) and the reports are now complete and standalone.

Thanks again for this contribution, and I hope to one day see another PR from you in Li.  Cheers and regards, jz

firebuggirl commented:
Hi @jzohrab,

Ah, no worries. Thank you for keeping me in the loop and thank you for letting me know about the new project over @ https://github.com/covidatlas/li . I look forward to checking it out!

jzohrab commented:
Cheers, hope you are doing well.  New project is interesting and there are
some useful things that would hopefully be fun to hack on.  Have a good
one! Z

El El sáb, jul. 11, 2020 a la(s) 1:43 p. m., Juliette <
notifications@github.com> escribió:

> Hi @jzohrab <https://github.com/jzohrab>,
>
> Ah, no worries. Thank you for keeping me in the loop and thank you for
> letting me know about the new project over @
> https://github.com/covidatlas/li . I look forward to checking it out!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/covidatlas/coronadatascraper/pull/543#issuecomment-657101813>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMPWDJ6JNC7LNEPJO24J6DR3CQFHANCNFSM4LXCFQMQ>
> .
>
